Dockerfile

```diff
@@ -0,0 +1,75 @@
+# syntax = docker/dockerfile:1
+ARG NODE_VERSION=20.0.0
+ARG SECRET_KEY
+ARG APP_DEBUG
+ARG TEST_SALEOR_API_URL
+FROM node:${NODE_VERSION}-slim as base
+
+# Adjust NODE_VERSION as desired
+
+ENV NODE_VERSION="$NODE_VERSION"
+
+
+LABEL fly_launch_runtime="Next.js"
+
+# Next.js app lives here
+WORKDIR /app
+
+# Install pnpm
+ARG PNPM_VERSION=8.12.0
+RUN npm install -g pnpm@$PNPM_VERSION
+
+
+# Throw-away build stage to reduce size of final image
+FROM base as build
+ARG SECRET_KEY
+ARG APP_DEBUG
+ARG TEST_SALEOR_API_URL
+ARG UPSTASH_URL
+ARG UPSTASH_TOKEN
+ARG APL
+
+ENV TEST_SALEOR_API_URL="$TEST_SALEOR_API_URL"
+ENV SECRET_KEY="$SECRET_KEY"
+ENV APP_DEBUG="$APP_DEBUG"
+ENV UPSTASH_URL="$UPSTASH_URL"
+ENV UPSTASH_TOKEN="$UPSTASH_TOKEN"
+ENV APL="${APL-upstash}"
```

Comment:
Will it work if I don't pass upstash creds, but set `APL=file`? 

APL should be injected agnostically from the outside world

Reply:
got it